syntax = "proto3";
package pt.tecnico.grpc;

message RegisterRequest {
  string name = 1; // name service
  string qualifier = 2;
  string address = 3;  // host:port
}

message RegisterResponse {
}

message LookupRequest {
  string name = 1;
  string qualifier = 2;
}

message LookupResponse {
  repeated string result = 1;
}

message DeleteRequest {
  string name = 1;
  string address = 2;
}

message DeleteResponse {
}

service NameServerService {
  rpc register(RegisterRequest) returns (RegisterResponse);
  rpc lookup(LookupRequest) returns (LookupResponse);
  rpc delete(DeleteRequest) returns (DeleteResponse);
}
